File: src/main/java/graveler/Mixins.java
package graveler;

import graveler.action.UpdateNeighborStress;
import graveler.action.UpdateStress;
import graveler.util.WorldExtensions;
import net.minecraft.core.BlockPos;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.level.Level;

public final class Mixins {
  private Mixins() {}

  public static void updateNeighbors(ServerLevel level, BlockPos pos) {
    if (level.isClientSide()) {
      return;
    }

    Scheduler scheduler = WorldExtensions.getScheduler(level);
    if (scheduler != null) {
      scheduler.schedule(new UpdateNeighborStress(pos));
    }
  }

  public static void onBlockAdded(Level level, BlockPos pos) {
    if (level.isClientSide()) {
      return;
    }

    Scheduler scheduler = WorldExtensions.getScheduler(level);
    if (scheduler != null) {
      scheduler.schedule(new UpdateStress(pos));
    }
  }
}

File: src/main/kotlin/graveler/Mixins.kt
@file:Suppress("unused")

package graveler

import graveler.action.UpdateNeighborStress
import graveler.action.UpdateStress
import graveler.util.scheduler
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IWorld
import net.minecraft.world.World

object Mixins {
  @JvmStatic
  fun updateNeighbors(world: IWorld, pos: BlockPos) {
    if (world.isRemote) return

    world.world.scheduler?.schedule(UpdateNeighborStress(pos))
  }

  @JvmStatic
  fun onBlockAdded(world: World, pos: BlockPos) {
    if (world.isRemote) return

    world.scheduler?.schedule(UpdateStress(pos))
  }
}
